replication/esecfse22/Dockerfile

```diff
@@ -32,7 +32,7 @@ WORKDIR /home/sherlock
 # Copy the compiled JAR file from the first stage into the second stage
 # Syntax: COPY --from=STAGE_ID SOURCE_PATH TARGET_PATH
 WORKDIR /home/sherlock/holmes
-COPY --from=0 /home/user/target/diffdetective-2.1.0-jar-with-dependencies.jar ./DiffDetective.jar
+COPY --from=0 /home/user/target/diffdetective-*-jar-with-dependencies.jar ./DiffDetective.jar
```

Comment:
I wonder if simplifying our workspace is worth introducing potential bugs here. As you mentioned, this will break in case there are multiple built versions, right? Maybe we could have two (instead of one) version files: the pom.xml, and a version.txt (or so) which is a file that contains just the version number as plain text. We can then use this version file in all the shell scripts. Just an idea.

Reply:
The `Dockerfile`s are not at risk for this problem because they build DiffDetective with a fresh `target` folder hence there will never be a second jar. The `.sh` scripts I changed are problematic.

An alternative would be to create a script which "parses" the `pom.xml` and outputs the version. Then we could use that script in the other scripts. Note that this would be a very lax parser because I want to avoid introducing a dependency on something like `xq` for developer machines.
```bash
# extracts the first version tag in pom.xml
sed -n '/<version/ {s/.*version>\(.*\)<\/version.*/\1/; p; q}' pom.xml
```

Adding a second "version" file kind of defeats the purpose in my opinion.

I also tested the Dockerfile right now to verify my claim: It doesn't actually build! The most relevant errors show that there is some Java version mismatch:
```
54.79 [ERROR] /home/user/src/main/java/org/variantsync/diffdetective/variation/diff/Projection.java:[10,39] cannot access org.variantsync.functjonal.list.FilteredMappedListView
54.79 [ERROR]   bad class file: /root/.m2/repository/org/variantsync/functjonal/1.0-SNAPSHOT/functjonal-1.0-SNAPSHOT.jar(/org/variantsync/functjonal/list/FilteredMappedListView.class)
54.79 [ERROR]     class file has wrong version 63.0, should be 61.0
54.79 [ERROR]     Please remove or make sure it appears in the correct subdirectory of the classpath.
```
The last Functjonal update was in 3ceefb4dcaf149ed277e4adc09eacbc81089cfcc. What motivated this commit?

Reply:
I guess the most important thing here is to end with a **robust** solution that does not easily break. If we have to overcomplicate things and make everything more fragile, I am fine with just having a bit of extra work when incrementing the version number: put the burden on the developer, not the user.

Apart from that, your `sed` solution looks good. It should be robust enough if reused within the other scripts.

It's bad news that the Docker file breaks. We should fix this asap on main because that is what FSE reviewers see. Could you have a look at it? Should we have the docker file build functjonal by itself? It seems I messed up with rebuilding the new version of functjonal. It is probably enough to build functjonal anew with an older version of Java. (The motivation for 3ceefb4 was exactly to fix this error. Previously, I had rebuild functjonal to introduce some new feature I needed somewhere but the jar was build with an even newer version.)

Reply:
@ibbem Which dockerfile did you test, when you reported the above error?

Reply:
I could repeat this error with the Dockerfile in `replication/esecfse22`.

Reply:
fixed in #137